scripts/check-repository-health.js

```diff
@@ -0,0 +1,328 @@
+  return termuxPackages;
+}
+
+async function getErrorsForArch(arch) {
+  const errors = [];
+  // This is a shell script which the maintainers can simply run on the aptly
+  // server to fix all errors that can the script was able to figure out the
+  // fix for.
+  const proposedAutomatedFixes = [];
+  const proposedManualFixes = [];
+  const termuxPackages = await getTermuxPackages(
+    arch,
+    errors,
+    proposedAutomatedFixes,
+    proposedManualFixes,
+  );
+  const aptPackages = new Map();
+  for (const path in repos) {
+    if (path == "pkg_format") continue;
+    const repo = repos[path];
+
+    // Get list of packages in apt repository and then add them to the map of all apt packages
+    const currentAptRepoPackages = await getAptPackages(
+      repo,
+      arch,
+      errors,
+      proposedAutomatedFixes,
+      proposedManualFixes,
+    );
+    for (const [pkgName, pkgInfo] of currentAptRepoPackages) {
+      // Check if the package should exist in this repository in the first place
+      if (termuxPackages.has(pkgName)) {
+        // Check if the package is in the correct repo
+        if (termuxPackages.get(pkgName).repo != pkgInfo.repo) {
+          // If not in correct repo, then it must be removed
+          errors.push(
+            `Package "${pkgName}" exists in "${pkgInfo.repo}" but should be in "${termuxPackages.get(pkgName).repo}"`,
+          );
+          proposedAutomatedFixes.push(
+            `aptly repo remove "${pkgInfo.repo}" "${pkgName} (=${pkgInfo.version}) {${arch}}"`,
+          );
+        } else {
+          // If it's in the correct repo, make sure it is the same version as we have in termux-packages
+          if (termuxPackages.get(pkgName).version != pkgInfo.version) {
+            errors.push(
+              `"${pkgName}" "${pkgInfo.version}" (in apt repository) != "${termuxPackages.get(pkgName).version}" (in termux-packages)`,
+            );
+            proposedManualFixes.push(
+              `The package "${pkgName}" in "${pkgInfo.repo}" may not be up to date on the apt repository or a bogus version may be present`,
+            );
+          }
+          aptPackages.set(pkgName, pkgInfo);
+        }
+      } else {
+        // If it's a static package, we need to repeat some of what we did earlier.
+        // -static packages are automatically generated, so they may cease to exist in newer versions when no static libraries exist in newer versions of base package
+        if (pkgName.endsWith("-static")) {
+          const basePkgName = pkgName.substring(
+            0,
+            pkgName.length - "-static".length,
+          );
+          if (termuxPackages.has(basePkgName)) {
+            // Check for TERMUX_PKG_NO_STATICSPLIT
+            if (!termuxPackages.get(basePkgName).mayHaveStaticSubpkg) {
+              errors.push(
+                `"${pkgName}" ${pkgInfo.version}: static package should not exist as parent package "${basePkgName}" has TERMUX_PKG_NO_STATICSPLIT=true`,
```

Comment:
This may need to be rechecked. There are packages that are `intentionally` subpackage with `-static` suffix but dont apply `TERMUX_PKG_NO_STATICSPLIT=true` or some other variants we do not know.